Source/WTF/wtf/ObjectIdentifier.h

```diff
@@ -26,37 +26,66 @@
 #pragma once
 
 #include <wtf/HashTraits.h>
+#include <wtf/UUID.h>
 #include <wtf/text/TextStream.h>
 #include <wtf/text/WTFString.h>
 
 namespace WTF {
 
+template<typename RawValue>
 struct ObjectIdentifierThreadSafeAccessTraits {
+};
+
+template<>
+struct ObjectIdentifierThreadSafeAccessTraits<uint64_t> {
     WTF_EXPORT_PRIVATE static uint64_t generateIdentifierInternal();
 };
 
+template<>
+struct ObjectIdentifierThreadSafeAccessTraits<UUID> {
+    WTF_EXPORT_PRIVATE static UUID generateIdentifierInternal();
+};
+
+template<typename RawValue>
 struct ObjectIdentifierMainThreadAccessTraits {
+};
+
+template<>
+struct ObjectIdentifierMainThreadAccessTraits<uint64_t> {
     WTF_EXPORT_PRIVATE static uint64_t generateIdentifierInternal();
 };
 
+template<>
+struct ObjectIdentifierMainThreadAccessTraits<UUID> {
+    WTF_EXPORT_PRIVATE static UUID generateIdentifierInternal();
+};
+
 // Extracted from ObjectIdentifierGeneric to avoid binary bloat.
+template<typename RawValue>
 class ObjectIdentifierGenericBase {
+};
+
+template<>
+class ObjectIdentifierGenericBase<uint64_t> {
 public:
+    using RawValue = uint64_t;
+
     bool isHashTableDeletedValue() const { return m_identifier == hashTableDeletedValue(); }
     bool isValid() const { return isValidIdentifier(m_identifier); }
 
-    uint64_t toUInt64() const { return m_identifier; }
+    RawValue toUInt64() const { return m_identifier; } // Use `toRawValue` instead.
```

Comment:
I don't understand the above comment.

Reply:
Also this seems wrong? Why have a RawValue return for a toUInt64 function?

Reply:
RawValue is just a type alias for `uint64_t` for this specialization, which promotes consistency between the specializations. The comment means that you should use `toRawValue` instead of `toUint64` since even though they both do the same thing, the former again helps consistency. Ideally, I would remove `toUint64`, but it's used in a lot of places, so this is the next best design.